% This is LLNCS.DEM the demonstration file of
% the LaTeX macro package from Springer-Verlag
% for Lecture Notes in Computer Science,
% version 2.4 for LaTeX2e as of 16. April 2010
%
\documentclass{llncs}
%
\usepackage{amssymb}
\usepackage{amsmath}
\usepackage{pdfpages}
\usepackage{graphicx}
\usepackage{paralist}
\usepackage{xspace}
\usepackage{paralist}
\usepackage{bm}
\usepackage{algorithm,algorithmic}
\newcommand{\numOTUs}{\ensuremath{L}}
\newcommand{\numSites}{\ensuremath{K}}
\newcommand{\numSp}{\ensuremath{M}}
\newcommand{\numObsStates}{\ensuremath{k}}
\newcommand{\virtPopSize}{\ensuremath{N}}
\newcommand{\polyProb}{\ensuremath{\phi}}
\newcommand{\pomoState}[3]{\ensuremath{\mathcal{S}_{#1#2}^{#3}}}
\newcommand{\triPomoState}[5]{\ensuremath{\mathcal{S}_{#1#2#3}^{#4,#5}}}
\newcommand{\quadPomoState}[3]{\ensuremath{\mathcal{S}_{\texttt{ACGT}}^{#1,#2,#3}}}
\newcommand{\Knorm}{\ensuremath{K}}
\newcommand{\Klmh}{\ensuremath{K_{LMH}}}
\newcommand{\pomo}{PoMo\xspace}
% from http://tex.stackexchange.com/a/33547
\newcommand{\appropto}{\mathrel{\vcenter{
              \offinterlineskip\halign{\hfil$##$\cr
                      \propto\cr\noalign{\kern2pt}\sim\cr\noalign{\kern-2pt}}}}}
\DeclareMathOperator*{\argmax}{\arg\!\max}
\usepackage{hyperref}
\hypersetup{backref, linkcolor=blue, citecolor=red, colorlinks=true, hyperindex=true}


\begin{document}
\title{A \pomo-like model for large-scale phylogenetics}
\titlerunning{POMO implementation}
\author{Mark T.~Holder\inst{1,2}\and Alexandros Stamatakis\inst{1,3}}
\authorrunning{Author 1 et al.} % abbreviated author list
%\tocauthor{Author 1, Author 2}
\institute{1. HITS gGmbH,\\2. Univ.~Kansas\\3. KIT}
%\email{\{Authors\}@h-its.org} \and Institute 2,\\ Address, Country\\
%\email{Author3@h-its.org}}
%% Eumerating parts in aligned environments %%
\newcommand{\enum}{\addtocounter{equation}{1}\tag{\theequation}}

\maketitle              % typeset the title of the contribution
\section {Motivation}
Large scale phylogenetic studies feature large numbers of species sampled with relatively few
    individuals sampled per species.
This sampling strategy make it difficult to examine the history of gene geneaologies with great precision, but 
    discordance between gene trees and species trees motivates the need for developing methods
    that recognize that polymorphism at the time of speciation can result in patterns of fixation
    that look like homoplasy when mapped onto the species tree.

The \pomo model of \cite{DeMaioSK2013} provides an elegant approximate method for dealing with polymorphism 
    on species trees.
However, the model is more general that needed for many phylogenetic purposes\footnote{or at least we {\em hope} that
    one need not implement such a rich model for the purpose of estimating trees in most cases.} and
    they formulate the model without the constraint of time reversibility.
The lack of time reversibility adds realism, but necessitates the use of rooted tree representations (furthermore
    the infer new parameters to describe the frequency of states at the root).

Here we develop a time-reversible that borrows ideas from the \pomo model, but is more amenable to 
    implementation in ExaML \cite{ExaMLInitial,ExaMLLatest}.
\section {Introduction}
DeMaio, Schl\"otterer, and Kosiol \cite{DeMaioSK2013} introduced a clever approach to dealing with polymorphism in the context of
    phylogenetic estimation.
Rather than explicitly treating the gene genealogy of a locus as a nuisance parameter, their polymorphism-aware
    phylogenetic model (\pomo) uses hidden state approach to model the frequency of residues for a site.

Input: Each of $\numOTUs$ OTUs is assumed to have sequence data from $\numSites$ sites and be mapped to one of $\numSp$ species.

The number of states of the data is $\numObsStates$. 
The model is described for DNA data ($\numObsStates=4$), but could be used for other small values of $\numObsStates$ if
    the process of fixation of a new state is assumed to be similiar to the genetic drift/selection as modelled by \pomo.

The model treats the evolution of each site as a continuous-time Markov process with a larger state space which
    includes states that correspond to the species being monomorphic for each of the $\numObsStates$ but als
    considers states representing polymorphic states in which 2 alleles occur at differing frequencies.
The continuous frequency of one allele in a population is binned for computational convenience.
This is done by imagining a virtual population of size $\virtPopSize$.

The process one residue replacing another would consists of:
\begin{compactenum}
\item A mutation such that the new allele has frequency $1/\virtPopSize$,
\item pass through each of the other polymorphic bins for this pair, $\{2/\virtPopSize, 3/\virtPopSize,\ldots,(\virtPopSize-1)/\virtPopSize, 1\}$
    where a frequency of 1 corresponds to the state that maps to the new allele's ``observable'' state being fixed.
\end{compactenum}
The drift process is reversible.

Evolutionary trajectories that involve a species being polymorphic for than one allele are clearly 
    biologically possible, but are prohibited to reduce the size of the state space.
    The state space of evolution in \pomo is: $\numObsStates + {\numObsStates \choose 2} \left(\virtPopSize - 1\right)$

\section{notation}
Simliar to this from  \cite{DeMaioSK2013} for convenience:
\begin{compactitem}
\item[$\bullet$] $\virtPopSize$ the virtual population size.
\item[$\bullet$] $\pomoState{I}{J}{K}$ is the state of displaying a frequency of the $I$ allele of $K/\virtPopSize$ and a frequency of $(N-K)/N$ for the $J$ allele.
\item[$\bullet$] $M_{IJ}^{i,k}$ the instantaneous rate of change from having the state corresponding to having $i$ copies of allele $I$ and $N-i$ copies of $J$ to having $k$ copies of $I$ and $N-k$ copies of $J$
\item[$\bullet$] $\polyProb$ will be used for what they called $\pi_{\mbox{pol}}$ for the sake of 
    legibility. This is the {\em a priori} probability of being in a polymorphic state at the root.
\item[$\bullet$] $s_I$ the selection coefficient of the allele with state $I$.
\item[$\bullet$] $\Knorm$ is a normalization constant (called $K_{\mbox{norm}}$ by \cite{DeMaioSK2013}).
\end{compactitem}

\section{A time-reversible form of \pomo}
The constraint of time-reversibility is that, $\forall X \neq Y$:
\begin{eqnarray}
    \pi_X q_{XY} = \pi_Y q_{YX}
\end{eqnarray}
were $\pi_X$ is the equilibrium state frequency of $X$, and $q_{XY}$ is the instantaneous
    rate of that an element of state $X$ tranitions to one of state $Y$.
This enables the refactoring into:
\begin{eqnarray}
    \pi_X q_{XY} & = & \pi_Y q_{YX} \\
    q_{XY} & = & r_{XY}\pi_Y \\
    q_{YX} & = & r_{XY}\pi_X
\end{eqnarray}
to emphasize the existence of a symmetric factor in the rate matrix, $r_{XY}$, multipled by the destination state frequency.

We can most clearly see that \pomo does not fit easily into a time-reversible framework by considering 
    the transition $M_{IJ}^{0,1}$ vs $M_{IJ}^{1,0}$. 
The former should be a very low rate constant reflecting the mutation rate.
The latter involves the fixation of an allele at very high frequency.
As Table S9 of  \cite{DeMaioSK2013} shows, these rates have different functional forms with respect to $N$ (the former is $N^2$; the latter is a function of $N^{-1}$).

Nevertheless, we can create \pomo-inspired reversible model by using
    $s_I= s_J$ for all $I$ and $J$, and then using the Eqn (4) of \cite{DeMaioSK2013}
    as the basis for the equilibrium state frequencies:
\begin{eqnarray}
\pi_{IJ}^{(i)} & = & \frac{\polyProb}{\Knorm }\left(
    \frac{\pi_J\mu_{JI}}{i} + \frac{\pi_I\mu_{IJ}}{N-i}\right) \hskip 4em \forall i: 0<i<N \label{stateFreqPoly}\\
    \Knorm  & = & \sum_I \sum_{J > I}\sum_{i=1}^{N-1} \left(
    \frac{\pi_J\mu_{JI}}{i} + \frac{\pi_I\mu_{IJ}}{N-i}\right) \\
    & = & \sum_I \sum_{J > I}\sum_{i=1}^{N-1} \left(
    \frac{N\pi_I\pi_J r_{IJ}}{i(N-i)}\right) \\
    & = & \sum_I \sum_{J > I}\left(N\pi_I\pi_J r_{IJ} \left[\sum_{i=1}^{N-1} \frac{1}{i(N-i)}\right]\right) \\
    & = &  N\left[\sum_{i=1}^{N-1} \frac{1}{i(N-i)}\right]\left(\sum_I \sum_{J > I}\pi_I\pi_J r_{IJ}\right) \label{KnormDef}
\end{eqnarray}
with the equilibrium frequency of being monomorphic for state $I$ is:
\begin{eqnarray}
    \pi_{IJ}^{(N)} = \pi_I(1-\polyProb) \label{monoStateFreq}
\end{eqnarray}
for any $J$,
as assumed for the root state frequencies (on page 2259 of \cite{DeMaioSK2013}).

Eqn(1) of \cite{DeMaioSK2013} is:
\begin{eqnarray}
 M_{IJ}^{i,i+1}  = \frac{i(1 + s_J - s_I)}{i(1 + s_J - s_I)+(N-i)}\times\frac{N-i}{N}
\end{eqnarray}
and their Eqn(2) is:
\begin{eqnarray}
 M_{IJ}^{i,i-1}  = \frac{N-i}{i(1 + s_J - s_I)+(N-i)}\times\frac{i}{N}
\end{eqnarray}
Setting $s_I=s_J$ simplifies these equations by changing $(1 + s_J - s_I)$ to 1 such that the become equal to each other:
\begin{eqnarray}
 M_{IJ}^{i,i+1}  = M_{IJ}^{i,i-1} & = & \frac{i(N-i)}{N^2}
\end{eqnarray}
with the corresponding elements of the $Q$-matrix for 
    transititions between polymorphic states are simply
    these elements multiplied by $N$:
\begin{eqnarray}
    Q_{IJ}^{i,i+1}  = Q_{IJ}^{i,i-1} & = & \frac{i(N-i)}{N} \label{polyQPair}
\end{eqnarray}

If we further think of the state frequencies being driven by a mutation rate composed of a product
    the nucleotide relative frequency and symmetric relative rate of interchange:
\begin{eqnarray}
 \mu_{IJ} & = &\pi_J r_{IJ} \\
 \mu_{JI} & = &\pi_I r_{IJ} 
\end{eqnarray}
then equation \ref{stateFreqPoly} becomes:
\begin{eqnarray}
\pi_{IJ}^{(i)} & = & \frac{\polyProb}{\Knorm }\left( \frac{\pi_J\pi_I r_{IJ}}{i} + \frac{\pi_I\pi_J r_{IJ}}{N-i}\right) \nonumber\\
     \pi_{IJ}^{(i)} & = & \frac{\polyProb}{\Knorm }\left( \frac{N\pi_I\pi_J r_{IJ}}{i(N-i)}\right) \label{polyStateFreqSimplified}
\end{eqnarray}

We can verify the time-reversibility of the $Q$-matrix in \ref{polyQPair} for the $i$ and $i+1$ pair:
\begin{eqnarray}
    \pi_{IJ}^{(i)} Q_{IJ}^{i,i+1} & = & \pi_{IJ}^{(i+1)} Q_{IJ}^{i+1,i} \nonumber \\
    \left( \frac{N\pi_I\pi_J r_{IJ}}{i(N-i)}\right) \left(\frac{i(N-i)}{N}\right) & = & \left( \frac{N\pi_I\pi_J r_{IJ}}{(i+1)(N-i-1)}\right)\left(\frac{(1+i)(N-i-1)}{N}\right) \nonumber
\end{eqnarray}

As with \pomo, for any transition involving a change of allele count $>1$, we set transition rates to be 0.

So to complete the $Q$-matrix we need the rates of entering the polymorphic state from a fixed state.
We can base the $1\leftrightarrow 0$, transition rates on the \pomo by adopting 
    the same transition rate for leaving the monomorphic state and
    then solving for the symmetric factor of the rate matrix, $r_{IJ}^{0,1}$
\begin{eqnarray}
    Q_{IJ}^{0,1} & = & N^2 \mu_{JI}  \label{toDiallelicQEl}\\
    Q_{IJ}^{0,1} & = & \pi_{IJ}^{(1)} r_{IJ}^{0,1} \nonumber\\
    \pi_{IJ}^{(1)} r_{IJ}^{0,1}& = & N^2 \mu_{JI}  \nonumber \\
    r_{IJ}^{0,1}& = & \frac{N^2 \mu_{JI}}{\pi_{IJ}^{(1)}}
\end{eqnarray}
Substituting the state frequency for the polymorphic state from Eqn (\ref{polyStateFreqSimplified}), we get:
\begin{eqnarray}
    r_{IJ}^{0,1} & = & \frac{\Knorm N^2 \mu_{JI}(N-1)}{\polyProb N\pi_I\pi_J r_{IJ}} \nonumber \\
    & = & \frac{\Knorm N^2 \pi_{I}r_{IJ}(N-1)}{\polyProb N\pi_I\pi_J r_{IJ}} \nonumber \\
                 & = & \frac{\Knorm N(N-1)}{\polyProb \pi_J} 
\end{eqnarray}
This provides us with a value for, $Q_{IJ}^{1,0}$, the fixation of state $I$, if we use Eqn(\ref{monoStateFreq}) for the equilibrium state frequency of the root state:
\begin{eqnarray}
    Q_{IJ}^{1,0} & = & \pi_{IJ}^{(0)} r_{IJ}^{0,1} \nonumber \\
                 & = & \left(\pi_J(1-\polyProb)\right)\left(\frac{\Knorm N(N-1)}{\polyProb \pi_J} \right) \nonumber \\
                 & = & \frac{\Knorm (1-\polyProb)N(N-1)}{\polyProb } \label{toMonomorphicQEl}
\end{eqnarray}

The non-zero elements of the transition rate matrix are thus given in equations \ref{polyQPair}, \ref{toDiallelicQEl}, and \ref{toMonomorphicQEl}.
The equilibrium (and root) state frequencies are given by equations \ref{polyStateFreqSimplified} and \ref{monoStateFreq}.

The free parameters are:
\begin{compactitem}
    \item[$\bullet$] the 5 relative mutation rates $\mu_{IJ}$ for all $I,J$ except the reference rate (usually the G$\leftrightarrow$T rate which is set to scale the rate matrix to the a standard expected rate of state-transitions).
    \item[$\bullet$] the 3 free base frequencies, $\pi_i$
    \item[$\bullet$] $\polyProb$. The MLE of $\polyProb$ could not be reliably estimated by \cite{DeMaioSK2013}, presumably because it only
    affected the likelihood via its affect on the weighting of states at the root.
    In our neutral variant, it appears in the transition rates at the boundary of the polymorphic/nonpolymorphic state pairs.
    Thus it will be a proxy for a scaled mutation rate, and should be more strongly identified by the data.
\end{compactitem}

The variables which appear in the model, but which are not parameters in the statistical sense:
\begin{compactitem}
    \item[$\bullet$] $\virtPopSize$, the virtual population size. the number of states scales linearly with this.
    This binning terms controls the accuracy of the approximation to continuous model 
    \item[$\bullet$] $\pi_{IJ}^{(i)}$ - a deterministic funtion of the parameters and \virtPopSize. See Eqns \ref{polyStateFreqSimplified} and \ref{monoStateFreq}.
    \item[$\bullet$] $\Knorm$ - a deterministic funtion of the parameters and \virtPopSize. See Eqn(\ref{KnormDef}).
\end{compactitem}

\subsection*{Summarizing $Q$ matrix and state frequency calculations}
\begin{itemize}
  \item by eqn \ref{polyQPair} for all transitions were $i$ and its neighboring are not $0$ or $N$ we have:
    $$Q_{IJ}^{i,i+1} = Q_{IJ}^{i,i-1} = \frac{i(N-i)}{N}$$
  \item by \ref{toDiallelicQEl}, introduction of new allele $I$ (from the state of fixed for $J$) is:
    $$Q_{IJ}^{0,1} = N^2 \pi_I r_{JI} $$ 
  \item by \ref{toMonomorphicQEl}, loss of the only copy of allele $I$ (leaving the population fixed for $J$) is: 
    $$Q_{IJ}^{1,0} =  \frac{\Knorm (1-\polyProb)N(N-1)}{\polyProb }$$
   where $\polyProb$ is a free parameter that is the stationary probability of being in a diallelic state, and (by eqn \ref{KnormDef}):
    $$\Knorm = N\left[\sum_{i=1}^{N-1} \frac{1}{i(N-i)}\right]\left(\sum_I \sum_{J > I}\pi_I\pi_J r_{IJ}\right) $$
  \item The diagonal elements are -1 times the sum of all elements in that row.
  \item all other elements of $Q$ are 0.
  \item by eqn \ref{monoStateFreq} the frequency of a monomorphic state: 
     $$\pi_{IJ}^{(N)} = \pi_{JI}^{(0)} = \pi_I(1-\polyProb) $$
    \item by eqn \ref{polyStateFreqSimplified} the frequency of a diallelic state: 
     $$\pi_{IJ}^{(i)} = \frac{\polyProb}{\Knorm }\left( \frac{N\pi_I\pi_J r_{IJ}}{i(N-i)}\right) $$
\end{itemize}

\section{Scaling the matrix}
Under neutrality, the mutation rate equals the substitution root.
However, the approximations involved in the version of \pomo described here, 
    it is not the case that flux out of a monomorphic state equals the flux
    of substitutions.

If we want the branch lengths to be expressed in terms of the expected substitutions
    per site, we will need to scale the rate matrix so that the
    rate of mutations that are expected to fix out of the 
    monomorphic states is 1.0.

High rates of mutation should increase the probability of a site being polymorphic.
Thus a theoretically justifiable model of among site rate variation would involve
    a different $\polyProb$ for each rate category.
However, that would lead to a rate-specific state-frequency spectrum, which would
    require summing over states at the root of the tree before summing over rate 
    categories.
This would not be hard to do, but I do not think that it is the way the ExaML works now.
So, the first attempt will keep $\polyProb$ constant and just scale the branch lengths.

\subsection{Scaling while treating $\polyProb$ as fixed}
The transitions that exit the monomorphic states represent the new mutations.
However, the result of subsequent transitions could result in the loss of the 
    new mutant nucleotide rather than a substitution.


If $i$ represents the number of copies of the mutant nucleotide in the diallelic
state (out of a virtual population size of $N$), then $\theta_i^{(N)}$ will
represent the probability that a character in this 
state will fix the novel nucleotide.

The general approach to scaling the matrix so that branches are scaled in terms of
    expected number of substitutions per site would be to calculate the unscaled
    flux of novel alleles that will fix:
\begin{eqnarray}
 U & = & \sum_{I \in \{A,C,G,T\}}\pi_{I\ast}^{(N)}\left[\sum_{J\neq I}Q_{IJ}^{N,N-1}\theta_1^{(N)}\right] \nonumber \\
 & = & \sum_{I \in \{A,C,G,T\}} \pi_I(1-\polyProb)\left[\sum_{J\neq I}N^2\pi_J r_{IJ}\theta_1^{(N)}\right] \nonumber\\
 & = & N^2(1-\polyProb)\theta_1^{(N)}\sum_{I \in \{A,C,G,T\}} \pi_I\left[\sum_{J\neq I}\pi_J r_{IJ}\right] \nonumber\\
  & = & N^2(1-\polyProb)\theta_1^{(N)}U_{GTR} \\
  U_{GTR} & = &\sum_{I \in \{A,C,G,T\}} \pi_I\left[\sum_{J\neq I}\pi_J r_{IJ}\right]
\end{eqnarray}
and then dividing every element in the $Q$ matrix by $U$.
$U_{GTR}$ is the scaler for the ``raw'' GTR matrix.

Note that the loss of an allele (either the loss of the new allele or the 
    loss of the ancestral allele) is associated with the same rate constant, which we will
    abbreviate:
    $$\ell = \frac{\Knorm (1-\polyProb)N(N-1)}{\polyProb }$$

For $N=2$ both of the exit transitions occur at the same rate, so $\theta_1^{(2)} = \frac{\ell}{2\ell} = \frac{1}{2}$.

For $N > 2$, the two transitions that can occur to a state with 1 copy of an allele are 
    loss of the allele (at rate $\ell$) and drift to have two copies of that allele.
The latter occurs at rate $$d = \frac{N-1}{N}$$
Thus, the probability that new mutant will fix is:
\begin{eqnarray}
    \theta_1^{(N)} = \frac{d\theta_2^{(N)}}{\ell + d} \\
    \theta_{N-1}^{(N)} = \frac{d\theta_{N-2}^{(N)} + \ell}{\ell + d}
\end{eqnarray}
for $N > 3$ and $i \neq 1$ and $i \neq N-1$:
\begin{eqnarray}
    \theta_{i}^{(N)} = \frac{\theta_{i-1}^{(N)} + \theta_{i+1}^{(N)}}{2}
\end{eqnarray}

This leads to the general statement for $N > 1$:
\begin{eqnarray}
    \theta_1^{(N)} = \frac{d}{2d + (N-2)\ell}
\end{eqnarray}

So:
\begin{eqnarray}
 U & = &  \frac{dN^2(1-\polyProb)U_{GTR}}{2d + (N-2)\ell} \\
  & = &  \frac{\frac{N-1}{N}N^2(1-\polyProb)U_{GTR}}{2\frac{N-1}N
                                                     + (N-2)\frac{\Knorm (1-\polyProb)N(N-1)}{\polyProb }} \\
  & = &  \frac{\polyProb(N-1)N^2(1-\polyProb)U_{GTR}}{2(N-1)\polyProb
                                                     + (N-2)\Knorm (1-\polyProb)N^2(N-1)} \\
  & = &  \frac{\polyProb N^2(1-\polyProb)U_{GTR}}{2\polyProb
                                                     + (N-2)\Knorm (1-\polyProb)N^2}
\end{eqnarray}

\subsection{Using the intrinsic, $\polyProb_I$ as a function of mutation rate}
Alternatively, we could force the loss of an allele rate to be equal to the rate that
    would be consistent with the fact that this event is simply a drift away from
    a state with 2 alleles.
If we solve for the $\polyProb_I$ associated with the unscaled mutation rate, then we
    can deal with a rate heterogeneity in a way that is more compatible with the 
    idea of a neutral model.

\begin{eqnarray}
    \frac{\Knorm (1-\polyProb_I)N(N-1)}{\polyProb_I} & = & \frac{N-1}{N} \nonumber \\
    \Knorm (1-\polyProb_I)N^2 & = & \polyProb_I \nonumber \\
    \Knorm N^2 & = & (1 + \Knorm N^2) \polyProb_I \nonumber \\
    \polyProb_I = \frac{\Knorm N^2}{1 + \Knorm N^2}
\end{eqnarray}

If we adopt this route, we'd probably want to have a ``raw'' values for $r_{ij}$ which 
    are then transformed by multiplying them by a $m/U_{GTR}$, where $m$ is the mean
    mutation rate.
Then for any rate, $m$, one could solve for $\polyProb_I$, and transform the
    final $Q$ matrix by multiplying each element by $\frac{m}{U}$.

\section{Transforming data}
We have sample $M$ species, and have $S_m$ individuals sampled from species $m$.
Assuming that each site is a random draw from the species current state is simple, though it ignores
    linkage disequilibrium between sites.

We can recode the $S_m$ sequences for species $m$ into a vector of tuples, $X_m^{(kk)}$ which 
    summarizes all of the sequence data for species $m$ at site $k$.

Each tuple has two or three elements.

The first element, $Z_m^{(k)}$, is an {\tt enum} style type designator which specifies what nucleotides are observed.
The valid facets for this {\tt enum} are: \texttt{A}, \texttt{C}, \texttt{G}, \texttt{T}, \texttt{AC}, \texttt{AG}, \texttt{AT}, \texttt{CG}, \texttt{CT}, and \texttt{GT}, \texttt{Missing}.
The first 4 describe sites that are monomorphic in a species.
The next six are the diallelic states.
The final state is used if none of the sequences from the species is scored for all sites.

If any species displays $>2$ states, then the likelihood will be 0 under our \pomo-like model.
So the processing should abort.

Let $S_m^{(k)}$ denote $S_m$ minus the number of sequences from species $m$ that have a missing data code at site $k$. 
So $S_m^{(k)}$ is the species sampling at this site.

If the type designator is monomorphic, then the other element of the tuple is $Y_m^{(k,0)}$, which is the number of sequences sampled from species $m$ that display the indicated state at site $k$.
$Y_m^{(k,0)}$ will be $S_m^{(k)}$.


If the type designator is diallelic, then there will be a two integers $Y_m^{(k,0)}$ and $Y_m^{(k,1)}$.
They will sum to $S_m^{(k)}$, and they will simply be the count of sequences for this species that display the first and second state specified by the type designator.

The phylogenetic tree will represent the $M$ species.

The conditional likelihood value (aka ``partial likelihood value'') for species $m$ and site $k$ will be constant through the analysis.
So the following values can be calculated upfront and stored in a constant cache;
with $C_m^{(k)}$ indicating the values for species $m$ and site $k$.

Up to this point, we have not established an ordering of the states of the \pomo-like model.

Here, we'll use the syntax $C_m^{(k)}[^{i}_{IJ}]$ will be used to indicate the element of 
    the tip partial likelihood that corresponds to a diallelic state for nucleotides $I$ and $J$
    with $i$ virtual individuals in state $J$ and $N-i$ individual in state $J$.

$C_m^{(k)}[I]$ will be used to indicate the element of 
    the tip partial likelihood that corresponds to a monomorphic state with all $\virtPopSize$
    individuals corresponding to state $I$.

We'll also use $\ast$ as a wildcard in this partial likelihood indexing.

\subsection*{$Z_m^{(k)}$ is \texttt{Missing}}
If $Z_m^{(k)}$ is \texttt{Missing}, then all elements of $C_m^{(k)}$ will be set to a probability of 1. So
\begin{compactitem}
    \item[$\bullet$] $C_m^{(k)}[\ast]=1.0$
\end{compactitem}

\subsection*{$Z_m^{(k)}$ is Diallelic}
If $Z_m^{(k)}$ is a diallelic code for with the first state denoted $Z_I$ and the second $Z_J$, then:
\begin{compactitem}
    \item[$\bullet$] $C_m^{(k)}[*] = 0$  (no monomorphic states explain the data)
    \item[$\bullet$] $C_m^{(k)}[^{\ast}_{IJ}] = 0$  if $I\neq Z_I$ or $J\neq Z_J$
    \item[$\bullet$] if $I = Z_I$ and $J = Z_J$, then
     $$C_m^{(k)}[^{i}_{IJ}] = {S_m^{(k)} \choose Y_m^{(k,0)}} (\virtPopSize - i)^{Y_m^{(k,0)}}i^{Y_m^{(k,1)}} \virtPopSize^{-S_m^{(k)}}$$
     which is simply the probability of the of the observed data from based on the binomial distribution.
     Note that the ``choose'' factor and the $ \virtPopSize^{-S_m^{(k)}}$ factors are constant for all values of $i$. So it could be dropped (because the likelihood need only be proportional to the probability of the data under the model).
\end{compactitem}


\subsection*{$Z_m^{(k)}$ is Monomorphic}
If $Z_m^{(k)}$ is a monomorphic code for the state $Z_I$ then:
\begin{compactitem}
    \item[$\bullet$] $C_m^{(k)}[I] = 0$ if $I\neq Z_I$
    \item[$\bullet$] $C_m^{(k)}[I] = 1$ if $I = Z_I$
    \item[$\bullet$] $C_m^{(k)}[^{\ast}_{IJ}] = 0$  if $I\neq Z_I$ and $J\neq Z_I$
    \item[$\bullet$] if $I = Z_I$
     $$C_m^{(k)}[^{i}_{IJ}] = \left(\frac{\virtPopSize - i}{\virtPopSize}\right)^{S_m^{(k)}}$$
    \item[$\bullet$] if $J = Z_I$, then
        $$C_m^{(k)}[^{i}_{IJ}] = \left(\frac{i}{\virtPopSize}\right)^{S_m^{(k)}}$$
\end{compactitem}
Note that the $ \virtPopSize^{-S_m^{(k)}}$ factors are constant for all values of $i$, but they 
     {\em cannot} be omitted because it does not occur in the $C_m^{(k)}[I]$ case.


\section{Some generalizations}
A generalization to multi-allelic sites is fairly straightforward, leading to a 14-state
    (for mono-, di-, and tri-allelic sites) a fully generic version with at least 15 states.
One question that arises in the abstract form of the model contemplated here, is whether mutation
    to a tri- or quad-allelic state can take place from any allele, or only from the most-frequent
    allele.

Here $\triPomoState{I}{J}{K}{N_I}{N_J}$ is the notation for $N_I$ copies of allele $I$, $N_J$ copies
    of allele $J$, and $\virtPopSize-N_I -N_J$ copies of allele $K$ when discussing tri-allelic states.
The most generic notation uses all four bases: $\quadPomoState{N_A}{N_C}{N_G}$ where $N_T$ would be $\virtPopSize-N_A - N_C - N_G$

If the different frequency bins of the states are interpreted as frequencies in a population, then
    it makes sense of the rate of transition to be simply reflect the frequency of the source allele 
    and the destination allele.
In a more abstract model (one with less of a tie to a sensible pop-gen perspective), one might view
    the state adjacent to the monomorphic state as primarily accommodating situations in which one
    allele is at very low frequency.
Any new mutation is likely to occur in the most frequent allele, thus the new mutation will be
    adding allele, never swapping one low-frequency allele for another.
This has to do with whether it is possible to transition directly from state $\pomoState{I}{J}{N-1}$ to
    $\pomoState{I}{K}{N-1}$ is allowed.
If these $N-1$ frequency states are viewed as proxies for cases of very low-frequency alleles, then 
    it is reasonable to prohibit these transitions and force either:
    $\pomoState{I}{J}{N-1} \leftrightarrow \pomoState{I}{\ast}{N} \leftrightarrow \pomoState{I}{K}{N-1}$
    or $\triPomoState{I}{J}{K}{N-1}{1} \leftrightarrow \triPomoState{I}{J}{K}{N-2}{1} \leftrightarrow \triPomoState{I}{J}{K}{N-1}{0}$

Table \ref{taxonomy} shows a taxonomy of \pomo-like models with the following column interpreations:
\begin{itemize}
    \item \textbf{Maximum \# alleles} present in species at any point: {\em 2}, {\em 3}, or {\em 4}
    \item \textbf{Treatment of polymorphism possible}: Only distinguishing between {\em variability} vs
    actually displaying {\em frequency} differences between states.
    \item \textbf{The source of the derivation of the rates}: base frequency {\em selection}, {\em neutrality}, or {\em abstract} (parameterizations that lack a clear population genetics interpretation).
    \item \textbf{Time Reversibility} {\em Yes} or {\em No}
    \item \textbf{Treatment of the $\virtPopSize -1$ and $1$ frequency states} As a {\em virtual} population size vs as a bin for very {\em low} frequency alleles.
    \item \textbf{$\virtPopSize$} the size of the virtual population, or $\ast$ if this is not internally consistent over the model
    \item \textbf{Total number of states}
\end{itemize}
\begin{table}
    \caption{A taxonomy of \pomo-like models.}\label{taxonomy}
        \begin{tabular}{r|c|c|c|c|c|r|r}
                           & max.~\# & polymorph.& source of & time & $(\virtPopSize -1)$ or 1  & & total \#\\
        Name               & alleles & treatment & rates     & rev. & treatment                 & $\virtPopSize $& states\\
        \hline
        DiVarAbsRevLow-2-10   & 2 & variability &abstract & Yes   & low & 2 & 10\\
        \hline
        QuadVarAbsRevLow-$\ast$-15 & 4 & variability &abstract & Yes   & low & $\ast$ & 15\\
        \hline
        \end{tabular}
    \begin{center}
    \end{center}
\end{table}

\begin{figure}
    \begin{center}
        %\includegraphics[scale=1]{images/10-state.pdf}
        \caption{DiVarAbsRevLow-2-10: Simplest \pomo-like model with just $\virtPopSize=2$. Diallelic states shown as smaller points. Monomorphic states as larger black circles.}\label{pomoDiVarAbsRevLow}
    \end{center}
\end{figure}

\begin{figure}
    \begin{center}
        \includegraphics[scale=1]{images/15-state.pdf}
        \caption{QuadVarAbsRevLow-$\ast$-15: Simplest general \pomo-like model with just $\virtPopSize=2$ for diallelic, 3 for triallelic and 1 state for the four-allele condition. Size so circle is used to denote the number of alleles (monomorphic are the largest). 15 total states.}\label{pomoQuadVarAbsRevLow}
    \end{center}
\end{figure}

\begin{figure}
    \begin{center}
        \includegraphics[scale=1]{images/35-state.pdf}
        \caption{QuadFreqAbsRevLow-4-35: Simplest general \pomo-like model with just $\virtPopSize=4$ that is internally consistent with repsect to $\virtPopSize$ and which models frequency of alleles (when in the diallelic condition and tri-allelic conditions). The blue arcs connect to the 4-allele state. The red arc  make other connections to the tri-allelic state)}\label{pomoQuadVarAbsRevLow}
    \end{center}
\end{figure}

\newpage
\section*{low-,mid-,high- binning}
See the lmh-pomo.tex and general-lmh-pomo.tex notes for some elaborations.

\bibliographystyle{splncs03}
\bibliography{pomo}




\end{document}

\begin{algorithm} \caption{}\label{}
\begin{algorithmic}
\end{algorithmic}
\end{algorithm}
